share/CMakeJS.cmake

```diff
@@ -0,0 +1,642 @@
+Check whether we have already been included (borrowed from CMakeRC)
+]=============================================================================]#
+# TODO: Decouple CMakeJS.cmake API version number from cmake-js version number...?
+set(_version 7.3.3)
+
+cmake_minimum_required(VERSION 3.15)
+cmake_policy(VERSION 3.15)
+include(CMakeParseArguments)
+
+if(COMMAND cmakejs_napi_addon_add_sources)
+    if(NOT DEFINED _CMAKEJS_VERSION OR NOT (_version STREQUAL _CMAKEJS_VERSION))
+        message(WARNING "More than one 'CMakeJS.cmake' version has been included in this project.")
+    endif()
+    # CMakeJS has already been included! Don't do anything
+    return()
+endif()
+
+set(_CMAKEJS_VERSION "${_version}" CACHE INTERNAL "Current 'CMakeJS.cmake' version. Used for checking for conflicts")
+
+set(_CMAKEJS_SCRIPT "${CMAKE_CURRENT_LIST_FILE}" CACHE INTERNAL "Path to current 'CMakeJS.cmake' script")
+
+# Default build output directory, if not specified with '-DCMAKEJS_BINARY_DIR:PATH=/some/dir'
+if(NOT DEFINED CMAKEJS_BINARY_DIR)
+    set(CMAKEJS_BINARY_DIR "${CMAKE_BINARY_DIR}")
+endif()
+
+message (STATUS "\n-- CMakeJS.cmake v${_CMAKEJS_VERSION}")
+
+#[=============================================================================[
+Internal helper (borrowed from CMakeRC).
+]=============================================================================]#
+function(_cmakejs_normalize_path var)
+    set(path "${${var}}")
+    file(TO_CMAKE_PATH "${path}" path)
+    while(path MATCHES "//")
+        string(REPLACE "//" "/" path "${path}")
+    endwhile()
+    string(REGEX REPLACE "/+$" "" path "${path}")
+    set("${var}" "${path}" PARENT_SCOPE)
+endfunction()
+
+#[=======================================================================[
+FindCMakeJs.cmake
+--------
+
+Find the native CMakeJs includes, source, and library
+
+(This codeblock typically belongs in a file named 'FindCMakeJS.cmake' for
+distribution...)
+
+This module defines
+
+::
+
+  CMAKE_JS_INC, where to find node.h, etc.
+  CMAKE_JS_LIB, the libraries required to use CMakeJs.
+  CMAKE_JS_SRC, where to find required *.cpp files, if any,
+  CMAKE_JS_EXECUTABLE, the cmake-js binary (global)
+  CMAKE_JS_NPM_PACKAGE, the cmake-js binary (local)
+
+]=======================================================================]#
+
+# CMAKE_JS_VERSION is defined on all platforms when calling from cmake-js.
+# By checking whether this var is pre-defined, we can determine if we are
+# running from an npm script (via cmake-js), or from CMake directly...
+
+if (NOT DEFINED CMAKE_JS_VERSION) # TODO: Unfortunately, this var is a little too close to 'CMAKE_VERSION' for comfort... Kitware may need it. Should be moved into 'CMAKEJS_VERSION'!
+
+    # ...and if we're calling from CMake directly, we need to set up some vars
+    # that our build step depends on (these are predefined when calling via npm/cmake-js).
+    if(VERBOSE)
+        message(STATUS "CMake Calling...")
+    endif()
```

Comment:
This if should be flattened out, as it will always be true following my suggestion above
```suggestion
```
(the end of the if needs manual deletion)